File: content/problems/0058-length-of-last-word/solutions/reverse-scan/solution.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int index = s.size() - 1; while (index >= 0 && s[index] == ' ') --index;
        int length = 0; while (index >= 0 && s[index] != ' ') { ++length; --index; }
        return length;
    }
};

File: content/problems/0058-length-of-last-word/solutions/reverse-scan/Solution.kt
class Solution {
    fun lengthOfLastWord(s: String): Int {
        var index = s.lastIndex; while (index >= 0 && s[index] == ' ') index--
        var length = 0; while (index >= 0 && s[index] != ' ') { length++; index-- }
        return length
    }
}
